source/extensions/filters/network/postgres_proxy/postgres_decoder.cc

```diff
@@ -271,6 +271,9 @@ bool DecoderImpl::onData(Buffer::Instance& data, bool frontend) {
     std::string message = "Unrecognized";
     if (f != nullptr) {
       auto msgParser = f();
+      // message_len_ specifies total message length including 4 bytes long
+      // "length" field. The length of message body is total length minus size
+      // of "length" field (4 bytes).
       msgParser->read(data, message_len_ - 4);
```

Comment:
s/message_len_ - 4/bytes_to_read/ ?

Reply:
Introduced new variable to skip carrying "-4".